prereise/gather/winddata/hrrr/calculations.py

```diff
@@ -0,0 +1,89 @@
+import pygrib
+from prereise.gather.winddata.hrrr.helpers import formatted_filename
+from powersimdata.utility.distance import angular_distance, ll2uv
+import numpy as np
+import pandas as pd
+from pandas import date_range
+from powersimdata.network.usa_tamu.constants.zones import id2abv
+
+from scipy.spatial import KDTree
+
+from prereise.gather.winddata.rap.power_curves import (
+    get_power,
+    get_state_power_curves,
+    get_turbine_power_curves,
+)
+
+U_COMPONENT_SELECTOR="U component of wind"
+V_COMPONENT_SELECTOR="V component of wind"
+
+
+def get_wind_data_lat_long(start_dt, directory):
+    gribs = pygrib.open(directory + formatted_filename(start_dt, "sfc", 0))
+    grib = next(gribs)
+    return grib.latlons()
+
+
+def find_closest_wind_grids(wind_farms, wind_data_lat_long):
+    grid_lats, grid_lons = wind_data_lat_long[0].flatten(), wind_data_lat_long[1].flatten()
+    grid_lat_lon_length = len(grid_lats)
+    assert grid_lat_lon_length == len(grid_lons)
+    grid_lat_lon_unit_vectors = [ll2uv(grid_lons[i], grid_lats[i]) for i in range(grid_lat_lon_length)]
+
+    tree = KDTree(grid_lat_lon_unit_vectors)
+
+    wind_farm_lats = wind_farms.lat.values
+    wind_farm_lons = wind_farms.lon.values
+
+    wind_farm_ct = len(wind_farms)
+
+    wind_farm_unit_vectors = [ll2uv(wind_farm_lons[i], wind_farm_lats[i]) for i in range(wind_farm_ct)]
+    _, indices = tree.query(wind_farm_unit_vectors)
+
+    return indices
+
+
+def calculate_pout(wind_farms, start_dt, end_dt, directory):
+    # Returns pandas dataframe in the following format:
+    #       wind_farm1  wind_farm2
+    # dt1        1        2
+    # dt2        3        4
+
+    id_target = wind_farms.index.values
```

Comment:
You can drop `values` here since the resulting `pandas.Index` array will be used in the `loc` (pandas internal  cuisine)